template/lib.typ

```diff
@@ -159,40 +167,70 @@
         []
       }
 
-      let level_numbering = if chapter.numbering == none {
+      //section number assignation
+      let item_numbering = if item.numbering == none {
           none
         } else {
-          context numbering(chapter.numbering, ..counter(heading).at(loc))
+          context numbering(item.numbering, ..counter(heading).at(loc))
         }
 
-      let entry = [#grid(columns: (20pt * (chapter.level - 1), 20pt, 1fr, auto), [], level_numbering, body, page_number)]
+      //data representation
 
-      let entry_spacing = if chapter.numbering != none and chapter.level == 1 { 10pt } else { 5pt }
+      //The first 'if' manages is for headings 
+      if item.at("level",default:none) != none{
+        let entry = [#grid(columns: (30pt * (item.level - 1), 30pt + 6pt * (item.level - 1), 1fr, auto), [], item_numbering, body, page_number)]
 
-      show grid: set block(above: entry_spacing, below: entry_spacing)
-
-      if chapter.level == 1 and chapter.numbering != none {
-        strong(entry)
+        let entry_spacing = if item.numbering != none and item.level == 1 { 25pt } else { 5pt }
+      
+        show grid: set block(above: entry_spacing)
+    
+       if (target == heading) or ( item.level <= 2 ){
+          if item.numbering != none {
+
+              if item.level == 1{
+                v(-1em)
+                strong(entry)
+                v(1em) 
+                } else {
+                v(-1em)
+                entry
+                v(1em) 
+               }
+
+          } else {
+             entry
+          }
+       }
```

Comment:
This is a fairly complex condition. What does the `target == heading` check translate to in this scenario?